### PR TITLE
satellite/repair: add timeout for repair download

### PR DESCRIPTION
**Pull request by VinozzZ**
What: 
- add timeout for repair download

Why:
- right now there's no timeout for download. A slow node could hold up the repair process

Please describe the tests:
 - Test 1:
 - Test 2:
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [x] Does the PR describe what changes are being made?
 - [x] Does the PR describe why the changes are being made?
 - [x] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [x] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [x] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [x] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [x] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [x] Does any documentation need updating?
 - [x] Do the database access patterns make sense?

### CODE REVIEW

File: satellite/repair/repairer/repairer.go [hunk 1]
```diff
@@ -29,6 +29,7 @@ type Config struct {
 	MaxRepair                     int           `help:"maximum segments that can be repaired concurrently" releaseDefault:"5" devDefault:"1"`
 	Interval                      time.Duration `help:"how frequently repairer should try and repair more data" releaseDefault:"1h" devDefault:"0h5m0s"`
 	Timeout                       time.Duration `help:"time limit for uploading repaired pieces to new storage nodes" default:"10m0s"`
+	DownloadTimeout               time.Duration `help:"time limit for downloading pieces from a node for repair" default:"2m0s"`
```
Comment (thepaul) [inline satellite/repair/repairer/repairer.go]: Can we bump this up to at least 5m? I'd rather try for a little bit longer to get the pieces we need before giving up and repairing them and adding to the garbage load. It's fairly reasonable for a node to be under high load or high network congestion for 2 minutes.

File: scripts/testdata/satellite-config.yaml.lock [hunk 2]
```diff
@@ -128,10 +128,10 @@ contact.external-address: ""
 # graceful-exit.overall-max-failures-percentage: 10
 
 # path to the certificate chain for this identity
-identity.cert-path: /root/.local/share/storj/identity/satellite/identity.cert
+identity.cert-path: /Users/yingrongzhao/Library/Application Support/Storj/Identity/Satellite/identity.cert
```
Comment (navillasa) [inline scripts/testdata/satellite-config.yaml.lock]: I don't think this should be changed :)

File: satellite/repair/repairer/segments.go [hunk 3]
```diff
@@ -51,7 +51,9 @@ type SegmentRepairer struct {
 func NewSegmentRepairer(
 	log *zap.Logger, metainfo *metainfo.Service, orders *orders.Service,
 	overlay *overlay.Service, dialer rpc.Dialer, timeout time.Duration,
-	excessOptimalThreshold float64, repairOverride int, satelliteSignee signing.Signee,
+	excessOptimalThreshold float64, repairOverride int,
+	downloadTimeout time.Duration,
```
Comment (navillasa) [inline satellite/repair/repairer/segments.go]: nit - seems kind of odd to have these two arguments on their own lines

Comment (navillasa) [reply satellite/repair/repairer/segments.go]: (lines 55 and 56)

**Pull request merged**